prov/efa/src/efa_shm.c

```diff
@@ -98,12 +98,27 @@ void efa_shm_info_create(const struct fi_info *app_info, struct fi_info **shm_in
 	int ret;
 	struct fi_info *shm_hints;
 
+	char *shm_provider;
+	if (rxr_env.use_sm2) {
+		shm_provider = "sm2";
+	} else {
+		shm_provider = "shm";
+	}
+
 	shm_hints = fi_allocinfo();
-	shm_hints->caps = FI_MSG | FI_TAGGED | FI_RECV | FI_SEND | FI_READ
-			   | FI_WRITE | FI_REMOTE_READ | FI_REMOTE_WRITE
-			   | FI_MULTI_RECV | FI_RMA | FI_SOURCE;
-	shm_hints->domain_attr->av_type = FI_AV_TABLE;
+	shm_hints->caps = app_info->caps;
+	shm_hints->caps &= ~FI_REMOTE_COMM;
```

Comment:
wait, is this cap is hints->caps or hints->domain_attr->caps?

Reply:
See L122

Reply:
Looks like putting in hints->caps is fine. This is how ompi mtl ofi does
```
hints->caps               = FI_TAGGED | FI_LOCAL_COMM | FI_REMOTE_COMM | FI_DIRECTED_RECV;
```

Reply:
I am ok as long as you have tested that shm can be picked up.

Reply:
Yes I did test this and verify SHM can still be picked up (with fabtests + osu benchmarks on OMPI 5).